virttest/test_setup.py

```diff
@@ -2052,15 +2052,20 @@ def disable_smt(params=None):
                                                server_user, server_pwd,
                                                r"[\#\$]\s*$")
         cmd_output = server_session.cmd_status_output(cmd)
-        if (cmd_output[0] == 0):
-            smt_output = cmd_output[1].strip()
-        else:
+        smt_output = cmd_output[1].strip()
+        if (cmd_output[0] != 0) and (smt_output not in SMT_DISABLED_STRS):
             server_session.close()
             raise exceptions.TestSetupFail("Couldn't get SMT of server: %s"
                                            % cmd_output[1])
     else:
-        smt_output = process.system_output(cmd, shell=True).strip()
-    if (smt_output != "SMT is off"):
+        try:
+            smt_output = process.system_output(cmd, shell=True).strip()
+        except process.CmdError as info:
+            smt_output = info.result.stderr.strip()
+            if smt_output not in SMT_DISABLED_STRS:
+                raise
```

Comment:
better to callout proper error, `exceptions.TestSetupFail...`

Reply:
Agreed and done.